src/models/krate.rs

```diff
@@ -371,28 +373,29 @@ impl Crate {
                 let config = &app.config;
                 match CrateOwnerInvitation::create(user.id, req_user.id, self.id, conn, config)? {
                     NewCrateOwnerInvitationOutcome::InviteCreated { plaintext_token } => {
-                        if let Ok(Some(recipient)) = user.verified_email(conn) {
-                            // Swallow any error. Whether or not the email is sent, the invitation
-                            // entry will be created in the database and the user will see the
-                            // invitation when they visit https://crates.io/me/pending-invites/.
-                            let email = OwnerInviteEmail {
-                                user_name: &req_user.gh_login,
-                                domain: &app.emails.domain,
-                                crate_name: &self.name,
+                        let email = user.verified_email(conn).ok().flatten().map(|recipient| {
+                            OwnerInviteEmail {
+                                recipient_email_address: recipient,
+                                user_name: req_user.gh_login.clone(),
+                                domain: app.emails.domain.clone(),
+                                crate_name: self.name.clone(),
                                 token: plaintext_token,
-                            };
+                            }
+                        });
 
-                            let _ = app.emails.send(&recipient, email);
-                        }
-
-                        Ok(format!(
+                        let msg = format!(
                             "user {} has been invited to be an owner of crate {}",
                             user.gh_login, self.name
-                        ))
+                        );
+
+                        Ok((msg, email))
```

Comment:
Instead of the model sending the invite email to the invitee, it now returns the email to the caller.

I kept the email struct generation here because querying for the user's email should probably happen within the same txn to preserve atomicity, and it was the smallest change to prevent the email spam.